functions/src/import/config.ts

```diff
@@ -0,0 +1,204 @@
+  }
+  if (value.length > sizeLimit) {
+    return `${name} must be at most ${sizeLimit} characters long`
+  }
+  if (!value.match(regex)) {
+    return `The ${name} does not match the regular expression provided`
+  }
+  return true
+}
+
+const validateBatchSize = (value: string) => {
+  return parseInt(value, 10) > 0
+}
+
+// Questions for the interactive user interface in command line.
+const questions = [
+  {
+    message: 'What is your Firebase project ID?',
+    name: 'project',
+    type: 'input',
+    validate: value =>
+      validateInput(
+        value,
+        'project ID',
+        FIRESTORE_VALID_CHARACTERS,
+        PROJECT_ID_MAX_CHARS
+      ),
+  },
+  {
+    message:
+      'What is the path of the Cloud Firestore Collection you would like to import from? ' +
+      '(This may, or may not, be the same Collection for which you plan to mirror changes.)',
+    name: 'sourceCollectionPath',
+    type: 'input',
+    validate: value =>
+      validateInput(
+        value,
+        'collection path',
+        FIRESTORE_VALID_CHARACTERS,
+        FIRESTORE_COLLECTION_NAME_MAX_CHARS
+      ),
+  },
+  {
+    message: 'Would you like to import documents via a Collection Group query?',
+    name: 'queryCollectionGroup',
+    type: 'confirm',
+    default: false,
+  },
+  {
+    message:
+      "What is the Uid of the Meilisearch index that you would like to use? (A index will be created if it doesn't already exist)",
+    name: 'index',
+    type: 'input',
+    validate: value =>
+      validateInput(
+        value,
+        'index',
+        MEILISEARCH_VALID_CHARACTERS,
+        MEILISEARCH_UID_MAX_CHARS
+      ),
+  },
+  {
+    message:
+      'How many documents should the import stream into Meilisearch at once?',
+    name: 'batchSize',
+    type: 'input',
+    default: 1000,
+    validate: validateBatchSize,
+  },
+  {
+    message:
+      'What is the host of the Meilisearch database that you would like to use? Example: http://localhost:7700.',
+    name: 'host',
+    type: 'input',
+  },
+  {
+    message:
+      'Which Meilisearch API key with permission to perform actions on indexes do you like to use? Both the private key and the master key are valid choices but we strongly recommend using the private key for security purposes.',
+    name: 'apiKey',
+    type: 'input',
+  },
+]
+
+export interface CliConfig {
+  projectId: string
+  sourceCollectionPath: string
+  queryCollectionGroup: boolean
+  batchSize: string
+  meilisearch: MeilisearchConfig
+}
+
+/**
+ * Parse the argument from the interactive or non-interactive command line.
+ */
+export async function parseConfig(): Promise<CliConfig> {
+  program.parse(process.argv)
+
+  const options = program.opts()
+  if (options.nonInteractive) {
+    if (
+      options.project === undefined ||
+      options.sourceCollectionPath === undefined ||
+      options.index === undefined ||
+      options.host === undefined ||
+      options.apiKey === undefined ||
```

Comment:
why are these not tested using `validateInput` ?

Reply:
`validateInput` is used with the interactive prompt so for every steps, we had the ability to ask for a valid input:
```
➜ npx firestore-meilisearch
? What is your Firebase project ID? 
>> Please supply a project ID
```
For the non-interactive prompt you will have the output helper:
```
npx firestore-meilisearch --project --source-collection-path movies --index books --batch-size 300 --non-interactive -H localhost:7700 -a masterKey
{"severity":"WARNING","message":"Warning, FIREBASE_CONFIG and GCLOUD_PROJECT environment variables are missing. Initializing firebase-admin will fail"}
Usage: firestore-meilisearch [options]

Options:
  --non-interactive                                      Parse all input from command line flags instead of prompting the caller. (default: false)
  -P, --project <project>                                Firebase Project ID for project containing the Cloud Firestore database.
  -s, --source-collection-path <source-collection-path>  The path of the Cloud Firestore Collection to import from. (This may, or may not, be the same Collection for which you plan to mirror changes.)
  -q, --query-collection-group [true|false]              Use 'true' for a collection group query, otherwise a collection query is performed.
  -i, --index <index>                                    The Uid of the index in Meilisearch to import to. (An index will be created if it doesn't already exist.)
  -b, --batch-size [batch-size]                          Number of documents to stream into Meilisearch at once. (default: 300)
  -H, --host <host>                                      The Host of your Meilisearch database. Example: http://localhost:7700.
  -a, --api-key <api-key>                                The Meilisearch API key with permission to perform actions on indexes. Both the private key and the master key are valid choices but we strongly recommend using the
                                                         private key for security purposes.
  -h, --help                                             display help for command
```